### PR TITLE
stop timer correctly

### PR DESCRIPTION
**Pull request by minggo**
Re: https://github.com/cocos/cocos-engine/issues/16653

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/core/scheduler.ts [hunk 1]
```diff
@@ -749,6 +750,33 @@ export class Scheduler extends System {
...
+        const element = this._hashForTimers[targetId];
+        const timers = element.timers;
+        if (!timers) {
+            return;
+        }
+
+        for (let i = 0, li = timers.length; i < li; i++) {
+            const timer = timers[i];
+            if (timer === timerToUnschedule) {
+                timers.splice(i, 1);
```
Comment (dumganhar) [inline cocos/core/scheduler.ts:770]: Doesn't it need to `CallbackTimer.put(timer);` after `timers.splice(i, 1)`?
I saw there was the logic of re-using the timer instance in `unschedule` function.

**Comment by dumganhar** [comment]
@minggo This pull request will cause another issue : https://forum.cocos.org/t/topic/159002/6
Test code:
```ts
import { _decorator, Component, game, Node } from 'cc';
const { ccclass, property } = _decorator;

@ccclass('test')
export class test extends Component {
    private count = 5;
    private times = 4;
    private indexCounter = 0;

    start() {
        this.test(1);
        this.test(2);
        this.test(3);
        this.testScheduleOnce();
    }

    private testScheduleOnce() {
        for (let i = 0; i < 5; i++) {
            this.doScheduleOnce(i);
        }
    }

    private doScheduleOnce(i: number) {
        let index = ++this.indexCounter;
        this.scheduleOnce(() => {
            console.log(`${i}  time    ` + index);
        });
    }

    private test(time: number) {
        this.scheduleOnce(()=>{
            console.log("time: ", time);
        });
    }

}



```

**Comment by kaxifakl** [comment]
3.8.2根据这个pr修复后，编辑器偶现这个报错
image

**Comment by minggo** [comment]
@kaxifakl 再根据 https://github.com/cocos/cocos-engine/pull/17215 修复一下就好了。

**Comment by kaxifakl** [comment]
> @kaxifakl 再根据 #17215 修复一下就好了。
这个pr也修过了，我这边查到报错是在编辑器中这个element为空，不知道什么原因导致的，暂时只有编辑器中某些预制体必现
image

**Comment by minggo** [comment]
382 没有问题是吗？能否建个 issue 上传下预制体？

**Comment by kaxifakl** [comment]
> 382 没有问题是吗？能否建个 issue 上传下预制体？

2个pr合并后出现的，3.8.2的编辑器，预制体文件在内网，暂时无法提供demo

**Pull request merged**